### ISSUE TITLE
Inference code

### ISSUE THREAD

**Issue by Naveen934** [huggingface/nanoVLM#58]
Hey please mention inference code like 

import torch
from PIL import Image
from transformers import AutoProcessor
from models.vision_language_model import VisionLanguageModel

# Load the pretrained model and processor (tokenizer + image transform)
model_name = "lusxvr/nanoVLM-222M"

# 1. Load the model
model = VisionLanguageModel.from_pretrained(model_name)
model.eval()

# 2. Load the processor (handles tokenization and image pre-processing)
processor = AutoProcessor.from_pretrained(model_name)

# 3. Prepare inputs
# Load your image
image = Image.open("your_image.jpg").convert("RGB")

# Your text input (e.g., a caption prompt)
text = "a photo of a cat"

# 4. Process inputs (image + text)
inputs = processor(text=[text], images=image, return_tensors="pt", padding=True)

# 5. Forward pass through the model
with torch.no_grad():
    outputs = model(**inputs)

# 6. Example output usage (depends on your downstream task)
# For example, get logits or similarity scores
logits = outputs.logits

print("Model output logits:", logits)

**Comment by lusxvr** [comment]
You can take a look at `generate.py` this has the code you are looking for! Let us know if you encounter any issues with it :)

**Comment by Naveen934** [comment]
Thanks for responce , its working well 

![Image](https://github.com/user-attachments/assets/c0b7ac2b-c8c2-460d-a82f-08f0c5e0b21c)
